myuw/dao/user_course_display.py

```diff
@@ -91,14 +92,19 @@ def _save_section_color(user, section, color_id):
     Store the color of the section in DB
     """
     section_label = section.section_label()
-    if UserCourseDisplay.exists_section_display(user, section_label):
-        _update_color(user, section_label, color_id)
```

Comment:
I eliminated the _update_color call at line#95 as the function _save_section_color is only called when the section_label is not in the existing_color_dict. Is it OK if the record is created by another process at almost the same time. For the same schedule the colors are deterministic.